convert2rhel/actions/conversion/preserve_only_rhel_kernel.py

```diff
@@ -50,34 +50,68 @@ def run(self):
             )
             return
 
-        # Check if kernel with same version is already installed.
+        # Check which of the kernel versions are already installed.
         # Example output from yum and dnf:
         #  "Package kernel-4.18.0-193.el8.x86_64 is already installed."
-        already_installed = re.search(r" (.*?)(?: is)? already installed", output, re.MULTILINE)
-        if already_installed:
-            rhel_kernel_nevra = already_installed.group(1)
-            non_rhel_kernels = pkghandler.get_installed_pkgs_w_different_fingerprint(
-                system_info.fingerprints_rhel, "kernel"
+        # When calling install, yum/dnf essentially reports all the already installed versions.
+        already_installed = re.findall(r" (.*?)(?: is)? already installed", output, re.MULTILINE)
+        # Get list of kernel pkgs not signed by Red Hat
+        non_rhel_kernels_pkg_info = pkghandler.get_installed_pkgs_w_different_fingerprint(
+            system_info.fingerprints_rhel, "kernel"
+        )
+        # Extract the NEVRA from the package object to a list
+        non_rhel_kernels = [pkghandler.get_pkg_nevra(kernel) for kernel in non_rhel_kernels_pkg_info]
+        rhel_kernels = [kernel for kernel in already_installed if kernel not in non_rhel_kernels]
+
+        # There is no RHEL kernel installed on the system at this point.
+        # Generally that would mean, that there is either only one kernel
+        # package installed on the system by the time of the conversion.
+        # Or none of the kernel packages installed is possible to be handled
+        # during the main transaction.
+        if not rhel_kernels:
+            info_message = (
+                "Conflict of kernels: The running kernel has the same version as the latest RHEL kernel. "
+                "The kernel package could not be replaced during the main transaction. "
+                "We will try to install a lower version of the package, "
+                "remove the conflicting kernel and then update to the latest security patched version."
+            )
+            loggerinst.info("\n%s" % info_message)
+            self.add_message(
+                level="INFO",
+                id="CONFLICT_OF_KERNELS",
+                title="Conflict of installed kernel versions",
+                description=info_message,
+            )
+            pkghandler.handle_no_newer_rhel_kernel_available()
+            kernel_update_needed = True
+
+        # In this case all kernel packages were already replaced during the main transaction
+        elif not non_rhel_kernels:
+            return
```

Comment:
Not at all required for this PR, but I think we could move this outside and turn into a if statement to skip early the rest of the code. 

```suggestion
        if not non_rhel_kernels:
            return
```

Reply:
Would look better for the workflow and easier to debug in future versions

Reply:
Reverted to `elif` in the end with the current implementation.
I've left an explanation in the comment.